core/shared/src/main/scala/fs2/Chunk.scala

```diff
@@ -306,30 +306,38 @@ abstract class Chunk[+O] extends Serializable with ChunkPlatform[O] with ChunkRu
       case _ => Chunk.ArraySlice(toArray, 0, size)
     }
 
-  /** Converts this chunk to a `java.nio.ByteBuffer`. */
+  /** Converts this chunk to a `java.nio.ByteBuffer`.
+    * @note that even "read-only" interaction with a `ByteBuffer` may increment its `position`,
+    * so this method should be considered as unsafely allocating mutable state.
+    */
   def toByteBuffer[B >: O](implicit ev: B =:= Byte): JByteBuffer =
     this match {
       case c: Chunk.ArraySlice[_] if c.values.isInstanceOf[Array[Byte]] =>
         JByteBuffer.wrap(c.values.asInstanceOf[Array[Byte]], c.offset, c.length)
       case c: Chunk.ByteBuffer =>
-        val b = c.buf.asReadOnlyBuffer
+        val b = c.buf.duplicate // share contents, independent position/limit
```

Comment:
`asReadOnlyBuffer` and `duplicate` are similar in that the returned buffer has independent position/limit.

The difference is that a read-only buffer denies access to its underlying `Array`, while a duplicated buffer does not. Being able to get at that array is an important optimization.